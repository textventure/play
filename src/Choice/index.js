import React, { Component } from 'react';
import Button from '@material-ui/core/Button';
import { withStyles } from '@material-ui/core/styles';
import history from '../helpers/history';

const styles = {
  button: {
    letterSpacing: 'initial',
    textAlign: 'left',
    textTransform: 'none',
  },
};

class Choice extends Component {
  /**
   * Selects choice.
   */
  onClick = () => {
    const { choiceId, currentId } = this.props;
    const search = history.location.search.replace(
      `id=${currentId}`,
      `id=${choiceId}`
    );
    history.push(search);
  };

  render() {
    const { children, className, classes } = this.props;
    return (
      <Button
        className={[classes.button, className].join(' ')}
        color="primary"
        onClick={this.onClick}
      >
        {children}
      </Button>
    );
  }
}

export default withStyles(styles)(Choice);
