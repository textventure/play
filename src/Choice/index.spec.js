import React from 'react';
import { shallow } from 'enzyme';
import { Button } from '@material-ui/core';
import history from '../helpers/history';
import Choice from '.';

let instance;
let props;
let wrapper;

jest.mock('../helpers/history', () => ({
  location: {},
  push: jest.fn(),
}));

afterAll(() => {
  jest.unmock('../helpers/history');
});

describe('when props={}', () => {
  beforeAll(() => {
    wrapper = shallow(<Choice />).dive();
  });

  it('renders correctly', () => {
    expect(wrapper.getElement()).toMatchSnapshot();
  });
});

describe('when props.children="foo"', () => {
  beforeAll(() => {
    props = { children: 'foo' };
    wrapper = shallow(<Choice {...props} />).dive();
  });

  it('renders text "foo"', () => {
    expect(wrapper.find(Button).children().text()).toBe('foo');
  });

  it('renders correctly', () => {
    expect(wrapper.getElement()).toMatchSnapshot();
  });
});

describe('when props.className="foo"', () => {
  beforeAll(() => {
    props = { className: 'foo' };
    wrapper = shallow(<Choice {...props} />).dive();
  });

  it('adds className to <Button>', () => {
    expect(wrapper.find(Button).prop('className')).toContain('foo');
  });

  it('renders correctly', () => {
    expect(wrapper.getElement()).toMatchSnapshot();
  });
});

describe('onClick', () => {
  beforeAll(() => {
    wrapper = shallow(<Choice />).dive();
    instance = wrapper.instance();
  });

  it('sets `onClick` on <Button>', () => {
    expect(wrapper.find(Button).prop('onClick')).toBe(instance.onClick);
  });

  it('calls `history.push` with props.choiceId', () => {
    history.location.search = '?id=currentId';
    wrapper.setProps({
      choiceId: 'choiceId',
      currentId: 'currentId',
    });
    instance.onClick();
    expect(history.push).toHaveBeenCalledWith('?id=choiceId');
  });
});
