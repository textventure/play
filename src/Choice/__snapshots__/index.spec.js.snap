// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when props.children="foo" renders correctly 1`] = `
<ForwardRef(WithStyles)
  className="Choice-button-1 "
  color="primary"
  onClick={[Function]}
>
  foo
</ForwardRef(WithStyles)>
`;

exports[`when props.className="foo" renders correctly 1`] = `
<ForwardRef(WithStyles)
  className="Choice-button-1 foo"
  color="primary"
  onClick={[Function]}
/>
`;

exports[`when props={} renders correctly 1`] = `
<ForwardRef(WithStyles)
  className="Choice-button-1 "
  color="primary"
  onClick={[Function]}
/>
`;
